truffleruby/src/main/ruby/core/truffle/cext.rb

```diff
@@ -1152,6 +1152,25 @@ def rb_rescue2(b_proc, data1, r_proc, data2, rescued)
     end
   end
 
+  def rb_struct_define_no_splat(name, attrs)
+    Struct.new(name, *attrs)
+  end
+
+  def rb_struct_aref(struct, index)
+    struct[index]
+  end
+
+  def rb_struct_aset(struct, index, value)
+    struct[index] = value
+  end
```

Comment:
These two could be done in C using `truffle_read` and `truffle_write`, which calls `[]` and `[]=` in Ruby automagically. But doesn't matter enough to change it now.